tests/serializers/test_natural.py

```diff
@@ -250,6 +253,51 @@ def fk_as_pk_natural_key_not_called(self, format):
         self.assertEqual(obj.object.pk, o1.pk)
 
 
+def natural_key_opt_out_test(self, format):
+    """
+    When a subclass of AbstractBaseUser opts out of natural key serialization
+    by returning an empty tuple, both FK and M2M relations serialize as
+    integer PKs and can be deserialized without error.
+    """
+    user1 = SubclassNaturalKeyOptOutUser.objects.create(email="user2@example.com")
+    user2 = SubclassNaturalKeyOptOutUser.objects.create(email="user3@example.com")
+
+    post = PostToOptOutSubclassUser.objects.create(
+        author=user1, title="Post 2 (Subclass Opt-out)"
+    )
+    post.subscribers.add(user1, user2)
+
+    user_data = serializers.serialize(format, [user1], use_natural_primary_keys=True)
+    post_data = serializers.serialize(format, [post], use_natural_foreign_keys=True)
+
+    list(serializers.deserialize(format, user_data))
+    deserialized_posts = list(serializers.deserialize(format, post_data))
+
+    post_obj = deserialized_posts[0].object
+    self.assertEqual(user1.email, post_obj.author.email)
+    self.assertEqual(
+        sorted([user1.email, user2.email]),
+        sorted(post_obj.subscribers.values_list("email", flat=True)),
+    )
+
+
+def deserialize_natural_key_then_opted_out(self, format):
+    """
+    Deserialization remains backward compatible:
+    data serialized with natural keys continues to load correctly
+    after the model opts out by returning empty tuple from natural_key().
+    """
+    user = NaturalKeyOptOut.objects.create(name="example")
+    serialized = serializers.serialize(format, [user], use_natural_primary_keys=True)
+
+    def optout_natural_key(self):
+        return ()
+
+    NaturalKeyOptOut.natural_key = optout_natural_key
```

Comment:
This test doesn't seem very realistic -- I don't think we want to set the expectation that you can change the natural key and load data you dumped before without problems. I think we can just remove it; then we don't have to worry about polishing this test. (I was about to suggest things like making sure state was restored after the test, since we're altering the class during the test, but it will be nice not to worry about that!)

If that leaves a coverage gap somewhere, let me know, and we can develop a different test.